### PR TITLE
0.57 upgrade Added steps to upgrade gradle version

### PR DESCRIPTION
**Pull request by DaniAkash**
Fixes react-native-community/react-native-releases#59

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -268,7 +269,12 @@ As mentioned a few times in the past, the core team is reviewing the repository
...
+4. Upgrading android gradle version to 4.4
+	1. In your project's `android/gradle/wrapper/gradle-wrapper.properties` file, change the `distributionUrl` to `https\://services.gradle.org/distributions/gradle-4.4-all.zip`
+	2. In `android/build.gradle` file add `google()` right above `jcenter()` in both `buildscript` and `allprojects` repositories. Then change Android build tools to version 3.1.3 `classpath 'com.android.tools.build:gradle:3.1.3'`
+	3. In `android/app/build.gradle` file update all your `compile` statements to be `implementation`, e.g. `implementation 'com.facebook.fresco:animated-gif:1.10.0'`
+	4. Do note that when running your app from within Android Studio, you may encounter `Missing Byte Code`errors. This is due to a known issue with version 3.1.x of the android tools plugin: [https://issuetracker.google.com/issues/72811718](https://issuetracker.google.com/issues/72811718). You'll need to disable Instant Run to get past this error.
```
Comment (turnrye) [inline CHANGELOG.md]: Missing a space:
```suggestion
	4. Do note that when running your app from within Android Studio, you may encounter `Missing Byte Code` errors. This is due to a known issue with version 3.1.x of the android tools plugin: [https://issuetracker.google.com/issues/72811718](https://issuetracker.google.com/issues/72811718). You'll need to disable Instant Run to get past this error.
```

**Comment by DaniAkash** [comment]
@turnrye have applied the changes...

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (kelset) [inline CHANGELOG.md]: can you remove this change?

File: CHANGELOG.md [hunk 3]
```diff
@@ -268,7 +269,12 @@ As mentioned a few times in the past, the core team is reviewing the repository
    ```
 
 3. Ensure that you have all the babel dependencies to version `^7.0.0` (you may also need to add `"babel-core": "7.0.0-bridge.0"` as a yarn resolution to ensure retro-compatibility)
-4. If you have a custom packager configuration via `rn-cli.config.js`, you probably need to update it to work with the updated Metro configuration structure (for full detail refer to Metro's [documentation](https://facebook.github.io/metro/docs/en/configuration)); here are some commonly encountered changes to `rn-cli.config.js`:
+4. Upgrading android gradle version to 4.4
```
Comment (kelset) [inline CHANGELOG.md]:
```suggestion
4. Upgrade the Android configuration:
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -268,7 +269,12 @@ As mentioned a few times in the past, the core team is reviewing the repository
...
+	1. In your project's `android/gradle/wrapper/gradle-wrapper.properties` file, change the `distributionUrl` to `https\://services.gradle.org/distributions/gradle-4.4-all.zip`
+	2. In `android/build.gradle` file add `google()` right above `jcenter()` in both `buildscript` and `allprojects` repositories. Then change Android build tools to version 3.1.3 `classpath 'com.android.tools.build:gradle:3.1.3'`
```
Comment (kelset) [inline CHANGELOG.md]: it looks like new RN projects use 3.1.4

File: CHANGELOG.md [hunk 5]
```diff
@@ -268,7 +269,12 @@ As mentioned a few times in the past, the core team is reviewing the repository
...
+	3. In `android/app/build.gradle` file update all your `compile` statements to be `implementation`, e.g. `implementation 'com.facebook.fresco:animated-gif:1.10.0'`
```
Comment (kelset) [inline CHANGELOG.md]: Can you make sure this is set as "optional"? Your project will work without it.

Comment (DaniAkash) [reply CHANGELOG.md]: without this change, it causes facebook/react-native#21309 issue in my system

Comment (turnrye) [reply CHANGELOG.md]: @DaniAkash based on @kelset's feedback, it seems like that may be a project specific or environment specific issue; are you open to making the recommended change?

Comment (kelset) [reply CHANGELOG.md]: I'm ok with not explicitly say to have it optional, can you fix the other issues at least so that we can merge it?

Comment (DaniAkash) [reply CHANGELOG.md]: Sure.

Comment (DaniAkash) [reply CHANGELOG.md]: @kelset it's done!

**Comment by kelset** [comment]
out it looks like there are some conflicts, @DaniAkash can you rebase and LMK when the PR is ready?

**Comment by turnrye** [comment]
Hey @DaniAkash, I'd love to merge this in but we need it to be rebased first. I'll leave the PR open another week or so but after that I'm going to close it as stale.

**Comment by DaniAkash** [comment]
@turnrye have rebased this pull request with the current master

**Comment by turnrye** [comment]
@DaniAkash thank you!

**Pull request merged**